app/models/smart_proxy.rb

```diff
@@ -91,11 +91,35 @@ def has_feature?(feature)
   def version
     result = {}
     begin
-      Timeout::timeout(20) do
```

Comment:
Seems that timeout is handled at the [ProxyApi::Resource](https://github.com/theforeman/foreman/blob/develop/lib/proxy_api/resource.rb#L8) and it is using Net::HTTP timeout which seems to be safer